src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -0,0 +1,466 @@
+use crate::commands::beta::generate::context::{CodeGenContext, DataModel};
+use crate::commands::beta::generate::result::{invalid_data_model_error, CodeGenResult};
+use crate::commands::beta::generate::utils::{Field, Import, Language};
+use convert_case::{Case, Casing};
+use ion_schema::isl::isl_constraint::{IslConstraint, IslConstraintValue};
+use ion_schema::isl::isl_type::IslType;
+use ion_schema::isl::isl_type_reference::IslTypeRef;
+use ion_schema::isl::IslSchema;
+use std::collections::HashMap;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+use tera::Context;
+
+pub(crate) struct CodeGenerator;
+
+impl CodeGenerator {
+    /// Maps the given type name to a native type based on programming language
```

Comment:
Naming things in the context of code generation gets very meta (and therefore very hairy) pretty quickly. Here's a naming convention that may help:

* `isl_type` => The type name that appeared in the ISL schema
* `target_type` => The _type_ that the `isl_type` maps to. (Rust `Vec`, Java `List`, etc).
* `target_kind` => The `target_type`'s type -- aligns with (but might not exactly match) the keyword used to define the type in the target language. (Rust `struct`/`enum`/primitive, Java `class`/`record`/primitive, etc).

```suggestion
    /// Maps the given ISL type name to a target type
```

"Native type" is a good phrase to use in contexts where the answer to the question "Native to what?" is obvious.

Reply:
Definitely. Thats a great suggestion to remove the ambiguity behind these similar naming concept.